src/Database/Orville/Internal/IndexDefinition.hs

```diff
@@ -35,3 +35,18 @@ simpleIndex name tableDef fields concurrently =
 
 indexFieldsBody :: [FieldDefinition] -> String
 indexFieldsBody fields = "(" ++ intercalate "," (map escapedFieldName fields) ++ ")"
+
+newtype IndexName = IndexName String deriving (Eq, Ord)
+
+indexNameToString :: IndexName -> String
+indexNameToString (IndexName i) = i
+
+stringToIndexName :: String -> IndexName
+stringToIndexName = either error id . safeStringToIndexName
+
+safeStringToIndexName :: String -> Either String IndexName
+safeStringToIndexName name =
+  if length name > 63 then
+    Left "Index greater than 63 characters"
```

Comment:
It might be nice to have more context here in the unlikely case that something does go wrong.

``` hs
Left $ "safeStringToIndexName: index name too long: " <> show name
```